// anef-ui/esm5/lib/navigation/navigation-bar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/navigation/navigation-bar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var NavigationBarComponent = /** @class */ (function () {
    function NavigationBarComponent() {
    }
    NavigationBarComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-navigation-bar',
                    template: "<div class=\"header-nav\" [ngClass]=theme>\n  <nav>\n    <p-menubar [model]=\"items\"></p-menubar>\n  </nav>\n</div>\n",
                    styles: [":host ::ng-deep .ui-menubar{background-color:#004990;border:0}:host ::ng-deep .ui-state-disabled,:host ::ng-deep .ui-state-disabled:focus,:host ::ng-deep .ui-state-disabled:hover{cursor:not-allowed!important;color:#656565;background-color:#004990;border-color:transparent}:host ::ng-deep .ui-state-active{background-color:#0f3c78;color:#fff;border-color:#d93e18;font-family:robotobold}:host ::ng-deep .light a{color:#256cb9}:host ::ng-deep .light .ui-menubar{background-color:#fff}:host ::ng-deep .light .ui-state-disabled,:host ::ng-deep .light .ui-state-disabled:focus,:host ::ng-deep .light .ui-state-disabled:hover{cursor:not-allowed!important;color:#c7c7c7;background-color:#fff;border-color:transparent}:host ::ng-deep .light .ui-state-active{background-color:#fff;color:#256cb9;border-color:#d93e18;font-family:robotobold}:host ::ng-deep .light a.is-active,:host ::ng-deep .light a:focus,:host ::ng-deep .light a:hover{background-color:#f5f5f5;color:#256cb9;border-color:#d93e18}:host ::ng-deep .active-tab{border-color:#d93e18}"]
                }] }
    ];
    /** @nocollapse */
    NavigationBarComponent.ctorParameters = function () { return []; };
    NavigationBarComponent.propDecorators = {
        theme: [{ type: Input }],
        items: [{ type: Input }]
    };
    return NavigationBarComponent;
}());
export { NavigationBarComponent };
if (false) {
    /** @type {?} */
    NavigationBarComponent.prototype.theme;
    /** @type {?} */
    NavigationBarComponent.prototype.items;
}
