// anef-ui/esm5/lib/information-modal/information-modal.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/information-modal/information-modal.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
var InformationModalComponent = /** @class */ (function () {
    function InformationModalComponent() {
        this.hideIconBloc = false;
        this.displayInfoChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    InformationModalComponent.prototype.onClose = /**
     * @return {?}
     */
    function () {
        this.displayInfoChange.emit(false);
    };
    /**
     * @return {?}
     */
    InformationModalComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.displayInfoChange.unsubscribe();
    };
    InformationModalComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-information-modal',
                    template: "<p-dialog\n  appendTo=\"body\"\n  [(visible)]=\"displayInfo\"\n  [modal]=\"true\"\n  [closable]=\"false\"\n  [responsive]=\"true\"\n  (onAfterHide)=\"onClose()\"\n  [style]=\"{width: '60rem', minWidth: '28rem'}\" [minY]=\"70\"\n  [baseZIndex]=\"10000\">\n  <p-header>\n    {{ title }}\n\n  </p-header>\n  <div class=\"msgBox info\" *ngIf=\"!hideIconBloc\">\n    <span class=\"fa fa-info-circle\"></span>\n    <p class=\"bold\">{{titleCheck}}</p>\n  </div>\n  <p class=\"bold\" *ngIf=\"textBold\">{{textBold | trim}}.</p>\n  <p>{{text | trim}}.</p>\n  <p *ngIf=\"facultativeText\">{{facultativeText}}</p>\n  <p-footer>\n    <button type=\"button\" pButton (click)=\"onClose()\"  label=\"{{labelBtn}}\"></button>\n  </p-footer>\n</p-dialog>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    InformationModalComponent.ctorParameters = function () { return []; };
    InformationModalComponent.propDecorators = {
        displayInfo: [{ type: Input }],
        title: [{ type: Input }],
        titleCheck: [{ type: Input }],
        text: [{ type: Input }],
        facultativeText: [{ type: Input }],
        labelBtn: [{ type: Input }],
        textBold: [{ type: Input }],
        hideIconBloc: [{ type: Input }],
        displayInfoChange: [{ type: Output }]
    };
    return InformationModalComponent;
}());
export { InformationModalComponent };
if (false) {
    /** @type {?} */
    InformationModalComponent.prototype.displayInfo;
    /** @type {?} */
    InformationModalComponent.prototype.title;
    /** @type {?} */
    InformationModalComponent.prototype.titleCheck;
    /** @type {?} */
    InformationModalComponent.prototype.text;
    /** @type {?} */
    InformationModalComponent.prototype.facultativeText;
    /** @type {?} */
    InformationModalComponent.prototype.labelBtn;
    /** @type {?} */
    InformationModalComponent.prototype.textBold;
    /** @type {?} */
    InformationModalComponent.prototype.hideIconBloc;
    /** @type {?} */
    InformationModalComponent.prototype.displayInfoChange;
}
