// anef-ui/esm5/lib/generic-modal/generic-modal.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/generic-modal/generic-modal.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Output, EventEmitter, Input, Component } from '@angular/core';
var GenericModalComponent = /** @class */ (function () {
    function GenericModalComponent() {
        this.modalWidth = 60;
        this.displayGenericModalChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    GenericModalComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.width = this.modalWidth + 'rem';
    };
    /**
     * @return {?}
     */
    GenericModalComponent.prototype.onClose = /**
     * @return {?}
     */
    function () {
        this.displayGenericModalChange.emit(false);
    };
    /**
     * @return {?}
     */
    GenericModalComponent.prototype.onSubmit = /**
     * @return {?}
     */
    function () {
        this.displayGenericModalChange.emit(true);
    };
    /**
     * @return {?}
     */
    GenericModalComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.displayGenericModalChange.unsubscribe();
    };
    GenericModalComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-generic-modal',
                    template: "<p-dialog\n    [class]=\"modalClass\"\n    [(visible)]=\"displayGenericModal\"\n    [modal]=\"true\"\n    [closable]=\"false\"\n    [responsive]=\"true\"\n    (onAfterHide)=\"onClose()\"\n    [style]=\"{width: width, minWidth: '28rem'}\"\n    [minY]=\"70\">\n    <p-header>\n        {{ title }}\n    </p-header>\n    <ng-content></ng-content>\n    <p-footer>\n        <button\n            *ngIf=\"!hideCloseBtn\"\n            type=\"button\"\n            class=\" button-with-neighbour\"\n            pButton\n            (click)=\"onClose()\"\n            label=\"{{closeBtn}}\"\n            [ngClass]=\"closeBtnClass ? closeBtnClass:'ui-button-secondary'\"\n        >\n        </button>\n        <button\n            *ngIf=\"!hideConfirmBtn\"\n            type=\"button\"\n            pButton\n            (click)=\"onSubmit()\"\n            label=\"{{confirmBtn}}\"\n            [ngClass]=\"confirmBtnClass ? confirmBtnClass:'ui-button-primary'\"\n            [disabled]=\"disableConfirmBtn\"\n            >\n        </button>\n    </p-footer>\n</p-dialog>\n",
                    styles: [""]
                }] }
    ];
    GenericModalComponent.propDecorators = {
        displayGenericModal: [{ type: Input }],
        title: [{ type: Input }],
        hideCloseBtn: [{ type: Input }],
        hideConfirmBtn: [{ type: Input }],
        closeBtn: [{ type: Input }],
        confirmBtn: [{ type: Input }],
        modalClass: [{ type: Input }],
        confirmBtnClass: [{ type: Input }],
        closeBtnClass: [{ type: Input }],
        disableConfirmBtn: [{ type: Input }],
        modalWidth: [{ type: Input }],
        displayGenericModalChange: [{ type: Output }]
    };
    return GenericModalComponent;
}());
export { GenericModalComponent };
if (false) {
    /** @type {?} */
    GenericModalComponent.prototype.displayGenericModal;
    /** @type {?} */
    GenericModalComponent.prototype.title;
    /** @type {?} */
    GenericModalComponent.prototype.hideCloseBtn;
    /** @type {?} */
    GenericModalComponent.prototype.hideConfirmBtn;
    /** @type {?} */
    GenericModalComponent.prototype.closeBtn;
    /** @type {?} */
    GenericModalComponent.prototype.confirmBtn;
    /** @type {?} */
    GenericModalComponent.prototype.modalClass;
    /** @type {?} */
    GenericModalComponent.prototype.confirmBtnClass;
    /** @type {?} */
    GenericModalComponent.prototype.closeBtnClass;
    /** @type {?} */
    GenericModalComponent.prototype.disableConfirmBtn;
    /** @type {?} */
    GenericModalComponent.prototype.modalWidth;
    /** @type {?} */
    GenericModalComponent.prototype.displayGenericModalChange;
    /** @type {?} */
    GenericModalComponent.prototype.width;
}
