// anef-ui/esm5/lib/breadcrumb/breadcrumb.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/breadcrumb/breadcrumb.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var BreadcrumbComponent = /** @class */ (function () {
    function BreadcrumbComponent() {
    }
    BreadcrumbComponent.decorators = [
        { type: Component, args: [{
                    selector: 'anef-breadcrumb',
                    template: "<div class=\"breadcrumb\">\n  <span class=\"label\">{{prefix}}</span>\n  <p-breadcrumb [model]=\"model\" [home]=\"home\" [style]=\"style\" [styleClass]=\"styleClass\"></p-breadcrumb>\n</div>\n",
                    styles: [".breadcrumb{display:flex;align-items:flex-end}"]
                }] }
    ];
    BreadcrumbComponent.propDecorators = {
        prefix: [{ type: Input }],
        model: [{ type: Input }],
        style: [{ type: Input }],
        styleClass: [{ type: Input }],
        home: [{ type: Input }]
    };
    return BreadcrumbComponent;
}());
export { BreadcrumbComponent };
if (false) {
    /** @type {?} */
    BreadcrumbComponent.prototype.prefix;
    /** @type {?} */
    BreadcrumbComponent.prototype.model;
    /** @type {?} */
    BreadcrumbComponent.prototype.style;
    /** @type {?} */
    BreadcrumbComponent.prototype.styleClass;
    /** @type {?} */
    BreadcrumbComponent.prototype.home;
}
