// anef-ui/esm2015/lib/forms/input-textarea-two-way-binding/input-textarea-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-textarea-two-way-binding/input-textarea-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { generateGuid } from '../../shared/guid';
export class InputTextareaTwoWayBindingComponent {
    constructor() {
        this.required = false;
        this.readonly = false;
        this.rows = 5;
        this.cols = 30;
        this.errors = [];
        this.ngModelChange = new EventEmitter();
        this.guid = generateGuid();
    }
}
InputTextareaTwoWayBindingComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line: component-selector
                selector: 'anef-input-textarea-two-way-binding',
                template: "<div class=\"p-grid nogutter\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" for=\"{{guid}}\">\n      {{label}} <span *ngIf=\"required && !readonly\" class=\"redish\">*</span> :\n    </label>\n\n    <textarea (ngModelChange)=\"ngModelChange.emit($event)\"\n              [(ngModel)]=\"model\"\n              [cols]=\"cols\"\n              [ngClass]=\"{'error': errors.length}\"\n              [readonly]=\"readonly\"\n              [rows]=\"rows\"\n              id=\"{{guid}}\"\n              name=\"{{guid}}\"\n              pInputTextarea\n              type=\"text\">\n    </textarea>\n\n    <div *ngIf=\"errors.length\" class=\"error redish\">\n      <div *ngFor=\"let error of errors\">\n        <p>\n          <i class=\"pi pi-info-circle\"></i>\n          <span>{{error}}</span>\n        </p>\n      </div>\n    </div>\n\n  </div>\n</div>\n",
                styles: [".textual{display:block}label{color:#656565;font-size:1.4rem}"]
            }] }
];
InputTextareaTwoWayBindingComponent.propDecorators = {
    model: [{ type: Input }],
    required: [{ type: Input }],
    readonly: [{ type: Input }],
    label: [{ type: Input }],
    rows: [{ type: Input }],
    cols: [{ type: Input }],
    errors: [{ type: Input }],
    ngModelChange: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.model;
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.required;
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.readonly;
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.rows;
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.cols;
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.errors;
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.ngModelChange;
    /** @type {?} */
    InputTextareaTwoWayBindingComponent.prototype.guid;
}
