// anef-ui/esm5/lib/forms/input-text/input-text.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-text/input-text.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// Angular
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { FormGroup } from '@angular/forms';
var InputTextComponent = /** @class */ (function () {
    function InputTextComponent() {
        this.MAX_LENGTH = 524288;
        this.required = false;
        this.readonly = false;
        this.capitalize = false;
        this.placeholder = '';
        this.maxLength = this.MAX_LENGTH;
        this.defaultDisplay = 'N/A';
        this.change = new EventEmitter();
        this.messages = {
            required: 'Ce champ est obligatoire',
            pattern: 'Le format n\'est pas correct',
            min: 'La valeur saisie est trop petite',
            max: 'La valeur saisie est trop grande',
            maxLength: 'La saisie dépasse le nombre de caractères autorisé'
        };
        this.textAfterReadOnly = '';
    }
    /**
     * @return {?}
     */
    InputTextComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.formGroup.get(this.controlName).valueChanges.subscribe((/**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            if (_this.capitalize) {
                /** @type {?} */
                var changed = value.charAt(0).toUpperCase() + value.slice(1);
                _this.formGroup.get(_this.controlName).setValue(changed, { emitEvent: false });
            }
        }));
    };
    /**
     * @return {?}
     */
    InputTextComponent.prototype.formContainsError = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].errors !== null &&
            (this.formGroup.controls[this.controlName].dirty || this.formGroup.controls[this.controlName].touched);
    };
    /**
     * @return {?}
     */
    InputTextComponent.prototype.toString = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].value || this.formGroup.controls[this.controlName].value === 0
            ? this.formGroup.controls[this.controlName].value : this.defaultDisplay;
    };
    InputTextComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-text',
                    template: "<div class=\"p-grid nogutter\" [formGroup]=\"formGroup\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" [htmlFor]=\"controlName\">\n      {{label}} <span class=\"redish\" *ngIf=\"required\">*</span> :\n    </label>\n\n    <ng-content select=\"[sub-label]\"></ng-content>\n\n    <input *ngIf=\"!readonly; else textual\"\n           [id]=\"controlName\"\n           type=\"text\"\n           pInputText\n           [formControlName]=\"controlName\"\n           [maxlength]=\"maxLength\"\n           [placeholder] = \"placeholder\">\n    <ng-template #textual>\n      <span class=\"textual\">{{ toString() }} &nbsp; {{ textAfterReadOnly }}</span>\n    </ng-template>\n\n    <ng-content select=\"[complement]\"></ng-content>\n\n    <div class=\"redish errors\" *ngIf=\"formContainsError()\">\n        <div *ngIf=\"formGroup.controls[controlName].errors.required\">\n          <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n          <span> {{messages.required}} </span>\n        </div>\n        <div *ngIf=\"formGroup.controls[controlName].errors.pattern\">\n          <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n          <span> {{messages.pattern}} </span>\n        </div>\n        <div *ngIf=\"formGroup.controls[controlName].errors.min && messages.min\">\n          <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n          <span> {{messages.min}} (minimum : {{formGroup.controls[controlName].errors.min.min}}) </span>\n        </div>\n        <div *ngIf=\"formGroup.controls[controlName].errors.max && messages.max\">\n          <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n          <span> {{messages.max}} (maximum : {{formGroup.controls[controlName].errors.max.max}}) </span>\n        </div>\n        <div *ngIf=\"formGroup.controls[controlName].errors.maxlength && messages.maxLength\">\n          <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n          <span> {{messages.maxLength}} (maximum : {{formGroup.controls[controlName].errors.maxlength.requiredLength}}) </span>\n        </div>\n        <ng-content select=\"[messages]\"></ng-content>\n    </div>\n  </div>\n</div>\n",
                    styles: [".textual{display:block}"]
                }] }
    ];
    InputTextComponent.propDecorators = {
        formGroup: [{ type: Input }],
        controlName: [{ type: Input }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        capitalize: [{ type: Input }],
        placeholder: [{ type: Input }],
        label: [{ type: Input }],
        maxLength: [{ type: Input }],
        defaultDisplay: [{ type: Input }],
        change: [{ type: Output }],
        messages: [{ type: Input }],
        textAfterReadOnly: [{ type: Input }]
    };
    return InputTextComponent;
}());
export { InputTextComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    InputTextComponent.prototype.MAX_LENGTH;
    /** @type {?} */
    InputTextComponent.prototype.formGroup;
    /** @type {?} */
    InputTextComponent.prototype.controlName;
    /** @type {?} */
    InputTextComponent.prototype.required;
    /** @type {?} */
    InputTextComponent.prototype.readonly;
    /** @type {?} */
    InputTextComponent.prototype.capitalize;
    /** @type {?} */
    InputTextComponent.prototype.placeholder;
    /** @type {?} */
    InputTextComponent.prototype.label;
    /** @type {?} */
    InputTextComponent.prototype.maxLength;
    /** @type {?} */
    InputTextComponent.prototype.defaultDisplay;
    /** @type {?} */
    InputTextComponent.prototype.change;
    /** @type {?} */
    InputTextComponent.prototype.messages;
    /** @type {?} */
    InputTextComponent.prototype.textAfterReadOnly;
}
