// anef-ui/esm5/lib/forms/edit-text-by-pencil/edit-text-by-pencil.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/edit-text-by-pencil/edit-text-by-pencil.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { FormControl } from '@angular/forms';
var EditTextByPencilComponent = /** @class */ (function () {
    function EditTextByPencilComponent() {
        this.isInputDisplayed = false;
    }
    /**
     * @return {?}
     */
    EditTextByPencilComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    EditTextByPencilComponent.prototype.onPencilClick = /**
     * @return {?}
     */
    function () {
        this.isInputDisplayed = !this.isInputDisplayed;
    };
    EditTextByPencilComponent.decorators = [
        { type: Component, args: [{
                    selector: 'edit-text-by-pencil',
                    template: "<div>\n  <div class=\"wrapper\">\n    <span *ngIf=\"labelForm\" class=\"content-text\">\n      {{ labelForm.value }}\n    </span>\n    <i\u00A0(click)=\"onPencilClick()\" class=\"fa fa-pencil\"></i>\n    <div>\n      <input pInputText *ngIf=\"isInputDisplayed\" type=\"text\" [formControl]=\"labelForm\" />\n    </div>\n    <div class=\"button-group\" *ngIf=\"isInputDisplayed\">\n      <span class=\"validate\">\n        <i class=\"fa fa-check-circle\"></i>\n        <span>VALIDER</span>\n      </span>\n      <span class=\"delete\">\n        <i class=\"fa fa-close\"></i>\n        <span>SUPPRIMER</span>\n      </span>\n    </div>\n  </div>\n\n</div>\n",
                    styles: [".fa.fa-pencil{color:#256cb9;float:right;cursor:pointer}.delete{color:#d13721;padding-left:.9rem}.validate{color:#107d11}.button-group span{opacity:1;cursor:pointer;text-align:left;font-weight:700;font-size:1.4rem;letter-spacing:0;text-transform:uppercase;font-family:robotoregular,\"Helvetica Neue\",sans-serif;margin-left:.3rem}"]
                }] }
    ];
    EditTextByPencilComponent.propDecorators = {
        labelForm: [{ type: Input }]
    };
    return EditTextByPencilComponent;
}());
export { EditTextByPencilComponent };
if (false) {
    /** @type {?} */
    EditTextByPencilComponent.prototype.labelForm;
    /** @type {?} */
    EditTextByPencilComponent.prototype.isInputDisplayed;
}
