// anef-ui/esm5/lib/forms/input-select-two-way-binding/input-select-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-select-two-way-binding/input-select-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { isEqual } from 'lodash';
import { generateGuid } from '../../shared/guid';
var InputSelectTwoWayBindingComponent = /** @class */ (function () {
    function InputSelectTwoWayBindingComponent() {
        this.error = null;
        this.required = false;
        this.showDeleteItemsButton = false;
        this.readonly = false;
        this.disabled = false;
        this.options = [];
        this.filter = false;
        this.ngModelChange = new EventEmitter();
        this.guid = generateGuid();
        this.colSize = '12';
    }
    Object.defineProperty(InputSelectTwoWayBindingComponent.prototype, "errors", {
        set: /**
         * @param {?} errors
         * @return {?}
         */
        function (errors) {
            if (!errors) {
                this.error = null;
            }
            else if (typeof (errors) === 'string') {
                this.error = [errors];
            }
            else {
                this.error = errors;
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    InputSelectTwoWayBindingComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.colSize = this.showDeleteItemsButton ? '11' : '12';
    };
    /**
     * @return {?}
     */
    InputSelectTwoWayBindingComponent.prototype.onModelChange = /**
     * @return {?}
     */
    function () {
        this.ngModelChange.emit(this.model);
    };
    /**
     * @return {?}
     */
    InputSelectTwoWayBindingComponent.prototype.toString = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (!this.options) {
            return '-';
        }
        /** @type {?} */
        var selectedOption = this.options.find((/**
         * @param {?} option
         * @return {?}
         */
        function (option) { return isEqual(_this.model, option.value); }));
        return selectedOption ? selectedOption.label : '-';
    };
    /**
     * @return {?}
     */
    InputSelectTwoWayBindingComponent.prototype.deleteItem = /**
     * @return {?}
     */
    function () {
        this.model = null;
        this.ngModelChange.emit(this.model);
    };
    InputSelectTwoWayBindingComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-select-two-way-binding',
                    template: "<div class=\"p-grid nogutter\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" for=\"{{guid}}\">\n      {{label}} <span *ngIf=\"required && !readonly\" class=\"redish\">*</span> :\n    </label>\n\n    <div class=\"input\">\n      <div *ngIf=\"!readonly; else textual\">\n        <div class=\"p-grid p-align-center\">\n          <p-dropdown\n            id=\"{{guid}}\"\n            name=\"{{guid}}\"\n            [placeholder]=\"placeholder\"\n            [(ngModel)]=\"model\"\n            (ngModelChange)=\"onModelChange()\"\n            [options]=\"options\"\n            [required]=\"required\"\n            [disabled]=\"disabled\"\n            class=\"p-col-{{colSize}}\"\n            [ngClass]=\"{\n              'error': error && error.length,\n              'fluid': !showDeleteItemsButton\n            }\"\n            [filter]=\"filter\">\n          </p-dropdown>\n\n          <span\n            (click)=\"deleteItem()\"\n            *ngIf=\"showDeleteItemsButton\"\n            class=\"delete-button\">\n              <i class=\"fa fa-close fa-lg icon\"></i>\n          </span>\n        </div>\n      </div>\n\n      <ng-template #textual>\n        <div class=\"textual\">{{ toString() }}</div>\n      </ng-template>\n    </div>\n\n    <div *ngIf=\"error && error.length\" class=\"error redish\">\n      <div *ngFor=\"let errorLine of error\">\n        <p>\n          <i class=\"pi pi-info-circle\"></i>\n          <span>{{errorLine}}</span>\n        </p>\n      </div>\n    </div>\n\n  </div>\n</div>\n",
                    styles: ["label{color:#656565;font-size:1.4rem}.delete-button{margin-bottom:7px;margin-left:5px;cursor:pointer;color:#d13721;font-size:1.2rem}p-dropdown{margin-top:-.5rem}p-dropdown.fluid{width:100%}.textual{padding-top:7px}"]
                }] }
    ];
    InputSelectTwoWayBindingComponent.propDecorators = {
        model: [{ type: Input }],
        required: [{ type: Input }],
        showDeleteItemsButton: [{ type: Input }],
        readonly: [{ type: Input }],
        disabled: [{ type: Input }],
        label: [{ type: Input }],
        placeholder: [{ type: Input }],
        options: [{ type: Input }],
        filter: [{ type: Input }],
        errors: [{ type: Input }],
        ngModelChange: [{ type: Output }]
    };
    return InputSelectTwoWayBindingComponent;
}());
export { InputSelectTwoWayBindingComponent };
if (false) {
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.error;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.model;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.required;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.showDeleteItemsButton;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.readonly;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.disabled;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.placeholder;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.options;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.filter;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.ngModelChange;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.guid;
    /** @type {?} */
    InputSelectTwoWayBindingComponent.prototype.colSize;
}
