// anef-ui/esm5/lib/forms/input-check/input-check.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-check/input-check.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// Angular
import { Component, Input } from '@angular/core';
import { FormGroup } from '@angular/forms';
var InputCheckComponent = /** @class */ (function () {
    function InputCheckComponent() {
        this.required = false;
        this.readonly = false;
        this.disableCheckbox = false;
        this.binary = false;
        this.messages = {
            required: 'Ce champ est obligatoire'
        };
    }
    /**
     * @return {?}
     */
    InputCheckComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (!this.name) {
            this.name = this.controlName;
        }
        if (this.disableCheckbox) {
            this.formGroup.controls[this.controlName].disable();
        }
    };
    Object.defineProperty(InputCheckComponent.prototype, "icon", {
        get: /**
         * @return {?}
         */
        function () {
            return this.option.icon || 'pi pi-check';
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    InputCheckComponent.prototype.formContainsError = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].errors !== null &&
            (this.formGroup.controls[this.controlName].dirty || this.formGroup.controls[this.controlName].touched);
    };
    /**
     * @return {?}
     */
    InputCheckComponent.prototype.toString = /**
     * @return {?}
     */
    function () {
        return this.option.label;
    };
    InputCheckComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-check',
                    template: "<div class=\"p-grid nogutter\" *ngIf=[formGroup] [formGroup]=\"formGroup\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" class=\"radio-label\" [htmlFor]=\"controlName\">\n      {{label}} <span class=\"redish\" *ngIf=\"required\">*</span> :\n    </label>\n\n    <ng-content select=\"[sub-label]\"></ng-content>\n\n    <p-checkbox *ngIf=\"!readonly; else textual\"\n                [name]=\"name\"\n                [value]=\"option.value\"\n                [label]=\"option.label\"\n                [formControlName]=\"controlName\"\n                [checkboxIcon]=\"icon\"\n                [binary]=\"binary\"\n                [styleClass]=\"option.styleClass\">\n    </p-checkbox>\n    <ng-template #textual>\n      <span class=\"textual\">{{toString()}}</span>\n    </ng-template>\n\n    <ng-content select=\"[complement]\"></ng-content>\n\n    <div class=\"redish errors\" *ngIf=\"formContainsError()\">\n      <div *ngIf=\"formGroup.controls[controlName].errors.required\">\n        <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n        <span> {{messages.required}} </span>\n      </div>\n\n      <ng-content select=\"[messages]\"></ng-content>\n    </div>\n  </div>\n</div>\n",
                    styles: ["label{width:100%}.no-col{flex-grow:initial;flex-basis:initial}.textual{display:block}"]
                }] }
    ];
    InputCheckComponent.propDecorators = {
        formGroup: [{ type: Input }],
        controlName: [{ type: Input }],
        name: [{ type: Input }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        disableCheckbox: [{ type: Input }],
        label: [{ type: Input }],
        option: [{ type: Input }],
        binary: [{ type: Input }],
        messages: [{ type: Input }]
    };
    return InputCheckComponent;
}());
export { InputCheckComponent };
if (false) {
    /** @type {?} */
    InputCheckComponent.prototype.formGroup;
    /** @type {?} */
    InputCheckComponent.prototype.controlName;
    /** @type {?} */
    InputCheckComponent.prototype.name;
    /** @type {?} */
    InputCheckComponent.prototype.required;
    /** @type {?} */
    InputCheckComponent.prototype.readonly;
    /** @type {?} */
    InputCheckComponent.prototype.disableCheckbox;
    /** @type {?} */
    InputCheckComponent.prototype.label;
    /** @type {?} */
    InputCheckComponent.prototype.option;
    /** @type {?} */
    InputCheckComponent.prototype.binary;
    /** @type {?} */
    InputCheckComponent.prototype.messages;
}
