// anef-ui/esm5/lib/modal-radio/modal-radio.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/modal-radio/modal-radio.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Output, Input } from '@angular/core';
var ModalRadioComponent = /** @class */ (function () {
    function ModalRadioComponent() {
        this.selectionChanged = new EventEmitter();
        this.selectionClosed = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ModalRadioComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.selectedIndex = this.preSelectedIndex;
    };
    /**
     * @return {?}
     */
    ModalRadioComponent.prototype.onClose = /**
     * @return {?}
     */
    function () {
        this.selectedIndex = this.preSelectedIndex;
        this.selectionClosed.emit(false);
    };
    /**
     * @return {?}
     */
    ModalRadioComponent.prototype.onSubmit = /**
     * @return {?}
     */
    function () {
        this.selectionChanged.emit(this.selectedIndex);
    };
    /**
     * @return {?}
     */
    ModalRadioComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.selectionChanged.unsubscribe();
        this.selectionClosed.unsubscribe();
    };
    /**
     * @param {?} i
     * @return {?}
     */
    ModalRadioComponent.prototype.setSelected = /**
     * @param {?} i
     * @return {?}
     */
    function (i) {
        this.selectedIndex = i;
    };
    ModalRadioComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-modal-radio',
                    template: "<p-dialog\n\t[class]=\"'modal-radio ' + modalClass\"\n\t[(visible)]=\"displayModalRadio\"\n\t[modal]=\"true\"\n\t[blockScroll]=\"true\"\n\t[closable]=\"false\"\n\t(onAfterHide)=\"onClose()\"\n\t[style]=\"{width: '60rem', minWidth: '28rem'}\"\n\t[minY]=\"70\"\n\t[baseZIndex]=\"10000\">\n\t<p-header>\n\t\t<div class=\"title\">\n\t\t\t{{ title }}\n\t\t</div>\n\t\t<div class=\"closer\" (click)=\"onClose()\"></div>\n\t</p-header>\n\t<div>\n\t\t<div *ngIf=\"topContentText\">\n\t\t\t{{ topContentText }}\n\t\t</div>\n\t\t<p-message *ngIf=\"warningMessage\" severity=\"warn\" text={{warningMessage}}></p-message>\n\t</div>\n\t<div class=\"p-grid p-justify-center list-bloc\">\n\t\t<div class=\"p-col-12\"\n\t\t\t*ngFor=\"let elem of radioList ; let index = index;\">\n\t\t\t<div class=\"p-grid p-align-center\">\n\t\t\t\t<div class=\"p-col-1\">\n\t\t\t\t\t<p-radioButton name=\"modalRadio\" [value]=\"index\" [(ngModel)]=\"selectedIndex\"></p-radioButton>\n\t\t\t\t</div>\n\t\t\t\t<div class=\"p-col-11 label\" (click)=\"setSelected(index)\">\n\t\t\t\t\t<ng-container *ngFor=\"let item of radioLabels\">\n\t\t\t\t\t\t<p *ngIf=\"elem[item.key]\">\n\t\t\t\t\t\t\t<strong>{{item.label}} : </strong> {{elem[item.key]}}\n\t\t\t\t\t\t</p>\n\t\t\t\t\t</ng-container>\n\t\t\t\t</div>\n\t\t\t</div>\n\t\t</div>\n\t</div>\n\t<p-footer>\n\t\t<button *ngIf=\"!hideCloseBtn\" type=\"button\" class=\" button-with-neighbour\" pButton (click)=\"onClose()\"\n\t\t\tlabel=\"{{closeBtn}}\" [ngClass]=\"closeBtnClass ? closeBtnClass:'ui-button-secondary'\">\n\t\t</button>\n\t\t<button *ngIf=\"!hideConfirmBtn\" type=\"button\" pButton (click)=\"onSubmit()\" label=\"{{confirmBtn}}\"\n\t\t\tclass=\"ui-button-primary\" [disabled]=\"selectedIndex == preSelectedIndex\">\n\t\t</button>\n\t</p-footer>\n</p-dialog>\n",
                    styles: [""]
                }] }
    ];
    ModalRadioComponent.propDecorators = {
        displayModalRadio: [{ type: Input }],
        title: [{ type: Input }],
        topContentText: [{ type: Input }],
        warningMessage: [{ type: Input }],
        hideCloseBtn: [{ type: Input }],
        hideConfirmBtn: [{ type: Input }],
        closeBtn: [{ type: Input }],
        confirmBtn: [{ type: Input }],
        modalClass: [{ type: Input }],
        confirmBtnClass: [{ type: Input }],
        closeBtnClass: [{ type: Input }],
        preSelectedIndex: [{ type: Input }],
        radioList: [{ type: Input }],
        radioLabels: [{ type: Input }],
        selectionChanged: [{ type: Output }],
        selectionClosed: [{ type: Output }]
    };
    return ModalRadioComponent;
}());
export { ModalRadioComponent };
if (false) {
    /** @type {?} */
    ModalRadioComponent.prototype.displayModalRadio;
    /** @type {?} */
    ModalRadioComponent.prototype.title;
    /** @type {?} */
    ModalRadioComponent.prototype.topContentText;
    /** @type {?} */
    ModalRadioComponent.prototype.warningMessage;
    /** @type {?} */
    ModalRadioComponent.prototype.hideCloseBtn;
    /** @type {?} */
    ModalRadioComponent.prototype.hideConfirmBtn;
    /** @type {?} */
    ModalRadioComponent.prototype.closeBtn;
    /** @type {?} */
    ModalRadioComponent.prototype.confirmBtn;
    /** @type {?} */
    ModalRadioComponent.prototype.modalClass;
    /** @type {?} */
    ModalRadioComponent.prototype.confirmBtnClass;
    /** @type {?} */
    ModalRadioComponent.prototype.closeBtnClass;
    /** @type {?} */
    ModalRadioComponent.prototype.preSelectedIndex;
    /** @type {?} */
    ModalRadioComponent.prototype.radioList;
    /** @type {?} */
    ModalRadioComponent.prototype.radioLabels;
    /** @type {?} */
    ModalRadioComponent.prototype.selectionChanged;
    /** @type {?} */
    ModalRadioComponent.prototype.selectionClosed;
    /** @type {?} */
    ModalRadioComponent.prototype.selectedIndex;
}
