// anef-ui/esm5/lib/tile/tile.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/tile/tile.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, HostBinding } from '@angular/core';
var TileComponent = /** @class */ (function () {
    function TileComponent() {
        this.tileClass = true;
    }
    TileComponent.decorators = [
        { type: Component, args: [{
                    selector: 'anef-tile',
                    template: "<a tabindex=\"0\">\n  <span class=\"tile-content p-grid no-grid-margin\">\n    <span class=\"p-col-3 p-sm-2 p-lg-12 p-col-align-center\">\n      <span class=\"fa p-align-center {{icon}}\"></span>\n    </span>\n    <span class=\"p-col-9 p-sm-10 p-lg-12 p-col-align-center\">\n      <span class=\"label\">\n        {{label}}\n      </span>\n    </span>\n  </span>\n</a>\n",
                    styles: [""]
                }] }
    ];
    TileComponent.propDecorators = {
        icon: [{ type: Input }],
        label: [{ type: Input }],
        tileClass: [{ type: HostBinding, args: ['class.ui-tile',] }]
    };
    return TileComponent;
}());
export { TileComponent };
if (false) {
    /** @type {?} */
    TileComponent.prototype.icon;
    /** @type {?} */
    TileComponent.prototype.label;
    /** @type {?} */
    TileComponent.prototype.tileClass;
}
