// anef-ui/esm5/lib/forms/input-select/input-select.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-select/input-select.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// Angular
import { Component, Input } from '@angular/core';
import { FormGroup } from '@angular/forms';
var InputSelectComponent = /** @class */ (function () {
    function InputSelectComponent() {
        this.required = false;
        this.readonly = false;
        this.messages = {
            required: 'Ce champ est obligatoire'
        };
        this.defaultDisplay = 'N/A';
        this.dropdownIcon = 'pi pi-chevron-down';
    }
    /**
     * @return {?}
     */
    InputSelectComponent.prototype.formContainsError = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].errors !== null &&
            (this.formGroup.controls[this.controlName].dirty || this.formGroup.controls[this.controlName].touched);
    };
    /**
     * @return {?}
     */
    InputSelectComponent.prototype.toString = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var found = null;
        if (this.formGroup && this.controlName && this.options) {
            /** @type {?} */
            var value_1 = this.formGroup.controls[this.controlName].value;
            found = (this.options && this.options.filter((/**
             * @param {?} option
             * @return {?}
             */
            function (option) { return option.value === value_1; }))[0]) || null;
        }
        return (found && found.label) || this.defaultDisplay;
    };
    InputSelectComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-select',
                    template: "<div class=\"p-grid nogutter\" [formGroup]=\"formGroup\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" [htmlFor]=\"controlName\">\n      {{label}} <span class=\"redish\" *ngIf=\"required\">*</span> :\n    </label>\n\n    <ng-content select=\"[sub-label]\"></ng-content>\n\n    <p-dropdown *ngIf=\"!readonly; else textual\"\n                [id]=\"controlName\"\n                name=\"controlName\"\n                [placeholder]=\"placeholder\"\n                [options]=options\n                [dropdownIcon]=\"dropdownIcon\"\n                [formControlName]=\"controlName\"></p-dropdown>\n    <ng-template #textual>\n      <span class=\"textual\">{{toString()}}</span>\n    </ng-template>\n\n    <ng-content select=\"[complement]\"></ng-content>\n\n    <div class=\"redish errors\" *ngIf=\"formContainsError()\">\n      <div *ngIf=\"formGroup.controls[controlName].errors.required\">\n        <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n        <span> {{messages.required}} </span>\n      </div>\n\n      <ng-content select=\"[messages]\"></ng-content>\n    </div>\n  </div>\n</div>\n",
                    styles: [".textual{display:block}"]
                }] }
    ];
    InputSelectComponent.propDecorators = {
        formGroup: [{ type: Input }],
        controlName: [{ type: Input }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        label: [{ type: Input }],
        placeholder: [{ type: Input }],
        options: [{ type: Input }],
        messages: [{ type: Input }],
        defaultDisplay: [{ type: Input }],
        dropdownIcon: [{ type: Input }]
    };
    return InputSelectComponent;
}());
export { InputSelectComponent };
if (false) {
    /** @type {?} */
    InputSelectComponent.prototype.formGroup;
    /** @type {?} */
    InputSelectComponent.prototype.controlName;
    /** @type {?} */
    InputSelectComponent.prototype.required;
    /** @type {?} */
    InputSelectComponent.prototype.readonly;
    /** @type {?} */
    InputSelectComponent.prototype.label;
    /** @type {?} */
    InputSelectComponent.prototype.placeholder;
    /** @type {?} */
    InputSelectComponent.prototype.options;
    /** @type {?} */
    InputSelectComponent.prototype.messages;
    /** @type {?} */
    InputSelectComponent.prototype.defaultDisplay;
    /** @type {?} */
    InputSelectComponent.prototype.dropdownIcon;
}
