// anef-ui/esm2015/lib/error-modal/error-modal.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/error-modal/error-modal.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
export class ErrorModalComponent {
    constructor() {
        this.displayErrorChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    onClose() {
        this.displayErrorChange.emit(false);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.displayErrorChange.unsubscribe();
    }
}
ErrorModalComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-error-modal',
                template: "<p-dialog class=\"modalAlert\"\n          appendTo=\"body\"\n          [showHeader]=\"false\"\n          [(visible)]=\"displayError\"\n          [modal]=\"true\"\n          [closable]=\"false\"\n          [responsive]=\"true\"\n          [style]=\"{width: '60rem', minWidth: '28rem'}\" [minY]=\"70\"\n          [baseZIndex]=\"10000\">\n\n  <div class=\"alertHeader\"\n    >{{ title }}\n  </div>\n  \n  <div class=\"msgBox alert\">\n    <span class=\"fa fa-exclamation-triangle\"></span>\n    <p>{{titleCheck}}</p>\n  </div>\n  <p class=\"bold\">{{textBold | trim}}.</p>\n  <p>{{text | trim}}.</p>\n  <p *ngIf=\"facultativeText\">{{facultativeText}}</p>\n  <p-footer>\n    <button type=\"button\" pButton (click)=\"onClose()\" label=\"{{labelBtn}}\"></button>\n  </p-footer>\n</p-dialog>\n",
                styles: [""]
            }] }
];
/** @nocollapse */
ErrorModalComponent.ctorParameters = () => [];
ErrorModalComponent.propDecorators = {
    displayError: [{ type: Input }],
    title: [{ type: Input }],
    titleCheck: [{ type: Input }],
    text: [{ type: Input }],
    facultativeText: [{ type: Input }],
    labelBtn: [{ type: Input }],
    textBold: [{ type: Input }],
    displayErrorChange: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    ErrorModalComponent.prototype.displayError;
    /** @type {?} */
    ErrorModalComponent.prototype.title;
    /** @type {?} */
    ErrorModalComponent.prototype.titleCheck;
    /** @type {?} */
    ErrorModalComponent.prototype.text;
    /** @type {?} */
    ErrorModalComponent.prototype.facultativeText;
    /** @type {?} */
    ErrorModalComponent.prototype.labelBtn;
    /** @type {?} */
    ErrorModalComponent.prototype.textBold;
    /** @type {?} */
    ErrorModalComponent.prototype.displayErrorChange;
}
