// anef-ui/esm5/lib/forms/input-textarea/input-textarea.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-textarea/input-textarea.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// Angular
import { Component, Input } from '@angular/core';
import { FormGroup } from '@angular/forms';
var InputTextareaComponent = /** @class */ (function () {
    function InputTextareaComponent() {
        this.MAX_LENGTH = 524288;
        this.required = false;
        this.readonly = false;
        this.rows = 5;
        this.cols = 30;
        this.maxLength = this.MAX_LENGTH;
        this.messages = {
            required: 'Ce champ est obligatoire',
            maxLength: 'Le nombre de caractères autorisés est dépassé'
        };
        this.defaultDisplay = 'N/A';
        this.spellCheck = false;
    }
    /**
     * @return {?}
     */
    InputTextareaComponent.prototype.formContainsError = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].errors !== null &&
            (this.formGroup.controls[this.controlName].dirty || this.formGroup.controls[this.controlName].touched);
    };
    /**
     * @return {?}
     */
    InputTextareaComponent.prototype.toString = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].value || this.defaultDisplay;
    };
    InputTextareaComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-textarea',
                    template: "<div class=\"p-grid nogutter\" [formGroup]=\"formGroup\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" [htmlFor]=\"controlName\">\n      {{label}} <span class=\"redish\" *ngIf=\"required\">*</span> :\n    </label>\n\n    <ng-content select=\"[sub-label]\"></ng-content>\n\n    <textarea *ngIf=\"!readonly; else textual\"\n              [id]=\"controlName\"\n              type=\"text\"\n              pInputTextarea\n              [rows]=\"rows\"\n              [cols]=\"cols\"\n              [maxLength]=\"maxLength\"\n              [formControlName]=\"controlName\"\n              [spellcheck]=\"spellCheck\"></textarea>\n    <ng-template #textual>\n      <span class=\"textual\">{{ toString() }}</span>\n    </ng-template>\n\n    <ng-content select=\"[complement]\"></ng-content>\n\n    <div class=\"redish errors\" *ngIf=\"formContainsError()\">\n      <div *ngIf=\"formGroup.controls[controlName].errors.required\">\n        <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n        <span> {{messages.required}} </span>\n      </div>\n      <div *ngIf=\"formGroup.controls[controlName].errors.maxlength\">\n        <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n        <span> {{messages.maxLength}} </span>\n      </div>\n\n      <ng-content select=\"[messages]\"></ng-content>\n    </div>\n  </div>\n</div>\n",
                    styles: [".textual{display:block}"]
                }] }
    ];
    InputTextareaComponent.propDecorators = {
        formGroup: [{ type: Input }],
        controlName: [{ type: Input }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        label: [{ type: Input }],
        rows: [{ type: Input }],
        cols: [{ type: Input }],
        maxLength: [{ type: Input }],
        messages: [{ type: Input }],
        defaultDisplay: [{ type: Input }],
        spellCheck: [{ type: Input }]
    };
    return InputTextareaComponent;
}());
export { InputTextareaComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    InputTextareaComponent.prototype.MAX_LENGTH;
    /** @type {?} */
    InputTextareaComponent.prototype.formGroup;
    /** @type {?} */
    InputTextareaComponent.prototype.controlName;
    /** @type {?} */
    InputTextareaComponent.prototype.required;
    /** @type {?} */
    InputTextareaComponent.prototype.readonly;
    /** @type {?} */
    InputTextareaComponent.prototype.label;
    /** @type {?} */
    InputTextareaComponent.prototype.rows;
    /** @type {?} */
    InputTextareaComponent.prototype.cols;
    /** @type {?} */
    InputTextareaComponent.prototype.maxLength;
    /** @type {?} */
    InputTextareaComponent.prototype.messages;
    /** @type {?} */
    InputTextareaComponent.prototype.defaultDisplay;
    /** @type {?} */
    InputTextareaComponent.prototype.spellCheck;
}
