// anef-ui/esm5/lib/confirm-modal/confirm-modal.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/confirm-modal/confirm-modal.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
var ConfirmModalComponent = /** @class */ (function () {
    function ConfirmModalComponent() {
        this.classIcon = 'fa fa-check-square-o';
        this.displayChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    ConfirmModalComponent.prototype.onClose = /**
     * @return {?}
     */
    function () {
        this.displayChange.emit(false);
    };
    /**
     * @return {?}
     */
    ConfirmModalComponent.prototype.onSubmit = /**
     * @return {?}
     */
    function () {
        this.displayChange.emit(true);
    };
    /**
     * @return {?}
     */
    ConfirmModalComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.displayChange.unsubscribe();
    };
    ConfirmModalComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-confirm-modal',
                    template: "<p-dialog\n  [(visible)]=\"display\"\n  [modal]=\"true\"\n  [closable]=\"false\"\n  [responsive]=\"true\"\n  (onAfterHide)=\"onClose()\"\n  [style]=\"{width: '60rem', minWidth: '28rem'}\" [minY]=\"70\"\n  [baseZIndex]=\"10000\">\n  <p-header>\n    {{ title }}\n  </p-header>\n  <div class=\"msgBox info\">\n    <span class=\"{{classIcon}} check-icon\"></span>\n    <p>{{titleIcon}}</p>\n  </div>\n  <p *ngIf=\"warning\" class=\"warning\">{{warning | trim}}</p>\n  <p class=\"bold\">{{textBold | trim}}</p>\n  <p>{{text | trim}}</p>\n  \n  <p-footer>\n    <button\n      type=\"button\"\n      class=\"button-with-neighbour ui-button-secondary\"\n      pButton\n      (click)=\"onClose()\"\n      label=\"{{closeBtn}}\"\n      >\n    </button>\n    <button\n      type=\"button\"\n      pButton\n      (click)=\"onSubmit()\"\n      label=\"{{confirmBtn}}\"\n      class=\"ui-button-primary\"\n      >\n    </button>\n  </p-footer>\n</p-dialog>\n",
                    styles: [".fa-check-square-o.check-icon{color:#0c6b0d!important}.warning{color:orange;background-color:#fbe5d6;text-align:start;padding:10px;margin-bottom:20px}"]
                }] }
    ];
    /** @nocollapse */
    ConfirmModalComponent.ctorParameters = function () { return []; };
    ConfirmModalComponent.propDecorators = {
        display: [{ type: Input }],
        title: [{ type: Input }],
        classIcon: [{ type: Input }],
        titleIcon: [{ type: Input }],
        textBold: [{ type: Input }],
        text: [{ type: Input }],
        warning: [{ type: Input }],
        closeBtn: [{ type: Input }],
        confirmBtn: [{ type: Input }],
        confirmBtnClass: [{ type: Input }],
        closeBtnClass: [{ type: Input }],
        displayChange: [{ type: Output }]
    };
    return ConfirmModalComponent;
}());
export { ConfirmModalComponent };
if (false) {
    /** @type {?} */
    ConfirmModalComponent.prototype.display;
    /** @type {?} */
    ConfirmModalComponent.prototype.title;
    /** @type {?} */
    ConfirmModalComponent.prototype.classIcon;
    /** @type {?} */
    ConfirmModalComponent.prototype.titleIcon;
    /** @type {?} */
    ConfirmModalComponent.prototype.textBold;
    /** @type {?} */
    ConfirmModalComponent.prototype.text;
    /** @type {?} */
    ConfirmModalComponent.prototype.warning;
    /** @type {?} */
    ConfirmModalComponent.prototype.closeBtn;
    /** @type {?} */
    ConfirmModalComponent.prototype.confirmBtn;
    /** @type {?} */
    ConfirmModalComponent.prototype.confirmBtnClass;
    /** @type {?} */
    ConfirmModalComponent.prototype.closeBtnClass;
    /** @type {?} */
    ConfirmModalComponent.prototype.displayChange;
}
