// anef-ui/esm5/lib/forms/input-multi-text-two-way-binding/input-multi-text-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-multi-text-two-way-binding/input-multi-text-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { generateGuid } from '../../shared/guid';
var InputMultiTextTwoWayBindingComponent = /** @class */ (function () {
    function InputMultiTextTwoWayBindingComponent() {
        this.MAX_LENGTH = 524288;
        this.error = null;
        this.type = 'text';
        this.required = false;
        this.readonly = false;
        this.capitalize = false;
        this.maxLength = this.MAX_LENGTH;
        this.ngModelChange = new EventEmitter();
        this.guid = generateGuid();
        this.value = [];
    }
    Object.defineProperty(InputMultiTextTwoWayBindingComponent.prototype, "model", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.value = value || [];
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(InputMultiTextTwoWayBindingComponent.prototype, "errors", {
        set: /**
         * @param {?} errors
         * @return {?}
         */
        function (errors) {
            if (!errors) {
                this.error = null;
            }
            else if (typeof (errors) === 'string') {
                this.error = [errors];
            }
            else {
                this.error = errors;
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    InputMultiTextTwoWayBindingComponent.prototype.onModelChange = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.capitalize) {
            this.value = this.value.map((/**
             * @param {?} element
             * @return {?}
             */
            function (element) { return _this._capitalize(element); }));
        }
        if (this.value[0] === '' && this.value.length === 1) {
            this.value = null;
        }
        this.ngModelChange.emit(this.value);
    };
    /**
     * @return {?}
     */
    InputMultiTextTwoWayBindingComponent.prototype.addItem = /**
     * @return {?}
     */
    function () {
        this.value.push('');
    };
    /**
     * @param {?} index
     * @return {?}
     */
    InputMultiTextTwoWayBindingComponent.prototype.deleteItem = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        this.value.splice(index, 1);
    };
    /**
     * @param {?} item
     * @return {?}
     */
    InputMultiTextTwoWayBindingComponent.prototype.toString = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        return item ? item : '-';
    };
    /**
     * @param {?} index
     * @param {?} item
     * @return {?}
     */
    InputMultiTextTwoWayBindingComponent.prototype.trackByFn = /**
     * @param {?} index
     * @param {?} item
     * @return {?}
     */
    function (index, item) {
        return index;
    };
    /**
     * @private
     * @param {?} text
     * @return {?}
     */
    InputMultiTextTwoWayBindingComponent.prototype._capitalize = /**
     * @private
     * @param {?} text
     * @return {?}
     */
    function (text) {
        return (text.charAt(0).toUpperCase() + text.slice(1));
    };
    InputMultiTextTwoWayBindingComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-multi-text-two-way-binding',
                    template: "<div class=\"p-grid nogutter\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" for=\"{{guid}}-0\">\n      {{label}} <span *ngIf=\"required && !readonly\" class=\"redish\">*</span> :\n    </label>\n\n    <div class=\"input\">\n      <div *ngIf=\"!readonly; else textual\">\n        <div class=\"p-grid p-align-center\">\n          <div [ngClass]=\"{\n            'p-col-11': value.length > 1,\n            'p-col-12': value.length <= 1\n          }\">\n            <input pInputText\n                   type=\"{{type}}\"\n                   id=\"{{guid}}-0\"\n                   name=\"{{guid}}-0\"\n                   [placeholder]=\"placeholder\"\n                   [maxlength]=\"maxLength\"\n                   [(ngModel)]=\"value[0]\"\n                   (ngModelChange)=\"onModelChange()\"\n                   [required]=\"required\"\n                   [ngClass]=\"{'error': error && error.length}\">\n          </div>\n\n          <span (click)=\"deleteItem(0)\"\n                *ngIf=\"value.length > 1\"\n                class=\"delete-button delete-{{guid}}-0\">\n              <i class=\"fa fa-close fa-lg icon\"></i>\n          </span>\n        </div>\n\n        <div *ngFor=\"let modelItem of value | slice:1; index as i; trackBy: trackByFn\"\n             class=\"p-grid p-align-center\">\n          <div [ngClass]=\"{\n            'p-col-11': value.length > 1,\n            'p-col-12': value.length <= 1\n          }\">\n            <input\n              pInputText\n              type=\"{{type}}\"\n              id=\"{{guid}}-{{i+1}}\"\n              name=\"{{guid}}-{{i+1}}\"\n              [maxlength]=\"maxLength\"\n              [(ngModel)]=\"value[i+1]\"\n              (ngModelChange)=\"onModelChange()\"\n              [ngClass]=\"{'error': error && error.length}\">\n          </div>\n\n          <span\n            (click)=\"deleteItem(i+1)\"\n            class=\"delete-button delete-{{guid}}-{{i+1}}\">\n              <i class=\"fa fa-close fa-lg icon\"></i>\n          </span>\n        </div>\n\n        <button (click)=\"addItem()\"\n                *ngIf=\"value[value.length - 1]\"\n                class=\"ui-button-borderless add-item\"\n                pButton>\n          {{addItemLabel}}\n        </button>\n      </div>\n\n      <ng-template #textual>\n        <ul class=\"item-readonly textual\">\n          <li *ngFor=\"let valueItem of value\">{{toString(valueItem)}}</li>\n          <li *ngIf=\"value.length === 0\">-</li>\n        </ul>\n      </ng-template>\n    </div>\n\n    <div *ngIf=\"error && error.length\" class=\"error redish\">\n      <div *ngFor=\"let errorLine of error\">\n        <p>\n          <i class=\"pi pi-info-circle\"></i>\n          <span>{{errorLine}}</span>\n        </p>\n      </div>\n    </div>\n\n  </div>\n</div>\n",
                    styles: [".item-readonly{list-style-type:none;padding:0;margin:0}label{color:#656565;font-size:1.4rem}.delete-button{margin-bottom:7px;margin-left:5px;cursor:pointer;color:#d13721;font-size:1.2rem}.input{margin-top:-1.3rem}.textual{margin-top:7px}"]
                }] }
    ];
    InputMultiTextTwoWayBindingComponent.propDecorators = {
        model: [{ type: Input, args: ['model',] }],
        type: [{ type: Input }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        capitalize: [{ type: Input }],
        label: [{ type: Input }],
        placeholder: [{ type: Input }],
        maxLength: [{ type: Input }],
        addItemLabel: [{ type: Input }],
        errors: [{ type: Input }],
        ngModelChange: [{ type: Output }]
    };
    return InputMultiTextTwoWayBindingComponent;
}());
export { InputMultiTextTwoWayBindingComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    InputMultiTextTwoWayBindingComponent.prototype.MAX_LENGTH;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.error;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.type;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.required;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.readonly;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.capitalize;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.placeholder;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.maxLength;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.addItemLabel;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.ngModelChange;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.guid;
    /** @type {?} */
    InputMultiTextTwoWayBindingComponent.prototype.value;
}
