// anef-ui/esm5/lib/forms/input-check-two-way-binding/input-check-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-check-two-way-binding/input-check-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { generateGuid } from '../../shared/guid';
var InputCheckTwoWayBindingComponent = /** @class */ (function () {
    function InputCheckTwoWayBindingComponent() {
        this.required = false;
        this.readonly = false;
        this.errors = [];
        this.ngModelChange = new EventEmitter();
        this.guid = generateGuid();
    }
    Object.defineProperty(InputCheckTwoWayBindingComponent.prototype, "icon", {
        get: /**
         * @return {?}
         */
        function () {
            return this.option.icon || 'pi pi-check';
        },
        enumerable: true,
        configurable: true
    });
    InputCheckTwoWayBindingComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-check-two-way-binding',
                    template: "<div class=\"p-grid nogutter\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" class=\"radio-label\" for=\"{{guid}}\">\n      {{label}} <span *ngIf=\"required && !readonly\" class=\"redish\">*</span> :\n    </label>\n\n    <div class=\"input\">\n      <p-checkbox (ngModelChange)=\"ngModelChange.emit($event)\"\n                  [(ngModel)]=\"model\"\n                  [checkboxIcon]=\"icon\"\n                  [label]=\"option.label\"\n                  [ngClass]=\"{'error': errors.length}\"\n                  [readonly]=\"readonly\"\n                  [styleClass]=\"option.styleClass\"\n                  id=\"{{guid}}\"\n                  name=\"{{guid}}\">\n      </p-checkbox>\n    </div>\n\n    <div *ngIf=\"errors.length\" class=\"error redish\">\n      <div *ngFor=\"let error of errors\">\n        <p>\n          <i class=\"pi pi-info-circle\"></i>\n          <span>{{error}}</span>\n        </p>\n      </div>\n    </div>\n\n  </div>\n</div>\n",
                    styles: ["label{width:100%;color:#656565;font-size:1.4rem}.no-col{flex-grow:initial;flex-basis:initial}.textual{display:block}"]
                }] }
    ];
    InputCheckTwoWayBindingComponent.propDecorators = {
        model: [{ type: Input }],
        name: [{ type: Input }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        label: [{ type: Input }],
        option: [{ type: Input }],
        errors: [{ type: Input }],
        ngModelChange: [{ type: Output }]
    };
    return InputCheckTwoWayBindingComponent;
}());
export { InputCheckTwoWayBindingComponent };
if (false) {
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.model;
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.name;
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.required;
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.readonly;
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.option;
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.errors;
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.ngModelChange;
    /** @type {?} */
    InputCheckTwoWayBindingComponent.prototype.guid;
}
