// anef-ui/esm5/lib/forms/input-multi-select-two-way-binding/input-multi-select-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-multi-select-two-way-binding/input-multi-select-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { isEqual } from 'lodash';
import { generateGuid } from '../../shared/guid';
var InputMultiSelectTwoWayBindingComponent = /** @class */ (function () {
    function InputMultiSelectTwoWayBindingComponent() {
        this.error = null;
        this.required = false;
        this.readonly = false;
        this.options = [];
        this.filter = false;
        this.limit = Infinity;
        this.ngModelChange = new EventEmitter();
        this.guid = generateGuid();
        this.value = [];
    }
    Object.defineProperty(InputMultiSelectTwoWayBindingComponent.prototype, "model", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.value = value || [];
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(InputMultiSelectTwoWayBindingComponent.prototype, "errors", {
        set: /**
         * @param {?} errors
         * @return {?}
         */
        function (errors) {
            if (!errors) {
                this.error = null;
            }
            else if (typeof (errors) === 'string') {
                this.error = [errors];
            }
            else {
                this.error = errors;
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    InputMultiSelectTwoWayBindingComponent.prototype.onModelChange = /**
     * @return {?}
     */
    function () {
        this.ngModelChange.emit(this.value);
    };
    /**
     * @return {?}
     */
    InputMultiSelectTwoWayBindingComponent.prototype.addItem = /**
     * @return {?}
     */
    function () {
        this.value.push(null);
    };
    /**
     * @param {?} index
     * @return {?}
     */
    InputMultiSelectTwoWayBindingComponent.prototype.deleteItem = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        this.value.splice(index, 1);
    };
    /**
     * @param {?} value
     * @return {?}
     */
    InputMultiSelectTwoWayBindingComponent.prototype.toString = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        /** @type {?} */
        var selectedOption = this.options.find((/**
         * @param {?} option
         * @return {?}
         */
        function (option) { return isEqual(value, option.value); }));
        return (selectedOption && selectedOption.label) ? selectedOption.label : '-';
    };
    InputMultiSelectTwoWayBindingComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-multi-select-two-way-binding',
                    template: "<div class=\"p-grid nogutter\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" for=\"{{guid}}-0\">\n      {{label}} <span *ngIf=\"required && !readonly\" class=\"redish\">*</span> :\n    </label>\n\n    <div [ngClass]=\"\" class=\"input\">\n      <div *ngIf=\"!readonly; else textual\">\n        <div class=\"p-grid p-align-center\">\n          <p-dropdown (ngModelChange)=\"onModelChange()\"\n                      [(ngModel)]=\"value[0]\"\n                      [filter]=\"filter\"\n                      [ngClass]=\"{\n                        'error': error && error.length,\n                        'fluid': value.length == 0\n                      }\"\n                      [options]=\"options\"\n                      [placeholder]=\"placeholder\"\n                      [required]=\"required\"\n                      class=\"p-col-11\"\n                      filter=\"true\"\n                      id=\"{{guid}}-0\"\n                      name=\"{{guid}}-0\">\n          </p-dropdown>\n\n          <span\n            (click)=\"deleteItem(0)\"\n            *ngIf=\"value.length\"\n            class=\"delete-button\">\n              <i class=\"fa fa-close fa-lg icon\"></i>\n          </span>\n        </div>\n\n        <div *ngFor=\"let valueItem of value | slice:1; index as i\" class=\"p-grid p-align-center additional-input\">\n          <p-dropdown (ngModelChange)=\"onModelChange()\"\n                      [(ngModel)]=\"value[i+1]\"\n                      [filter]=\"filter\"\n                      [ngClass]=\"{'error': error && error.length}\"\n                      [options]=\"options\"\n                      [placeholder]=\"placeholder\"\n                      [required]=\"required\"\n                      class=\"p-col-11\"\n                      filter=\"true\"\n                      id=\"{{guid}}-{{i+1}}\"\n                      name=\"{{guid}}-{{i+1}}\">\n          </p-dropdown>\n\n          <span\n            (click)=\"deleteItem(i+1)\"\n            class=\"delete-button\">\n              <i class=\"fa fa-close fa-lg icon\"></i>\n          </span>\n        </div>\n\n        <button (click)=\"addItem()\"\n                *ngIf=\"value[value.length - 1] && value.length < limit\"\n                class=\"ui-button-borderless add-item\"\n                pButton>\n          {{addItemLabel}}\n        </button>\n      </div>\n\n      <ng-template #textual>\n        <ul class=\"item-readonly textual\">\n          <li>{{ toString(value[0]) }}</li>\n          <li *ngFor=\"let valueItem of value | slice:1;\">{{ toString(valueItem) }}</li>\n        </ul>\n      </ng-template>\n    </div>\n\n    <div *ngIf=\"error && error.length\" class=\"error redish\">\n      <div *ngFor=\"let errorLine of error\">\n        <p>\n          <i class=\"pi pi-info-circle\"></i>\n          <span>{{errorLine}}</span>\n        </p>\n      </div>\n    </div>\n\n  </div>\n</div>\n",
                    styles: [".item-readonly{list-style-type:none;padding:0;margin:0}.additional-input{margin-top:1rem}label{color:#656565;font-size:1.4rem}.delete-button{cursor:pointer;color:#d13721;font-size:1.2rem;margin-bottom:7px;margin-left:5px}p-dropdown{margin-top:-.5rem}p-dropdown.fluid{width:100%}.textual{margin-top:7px}"]
                }] }
    ];
    InputMultiSelectTwoWayBindingComponent.propDecorators = {
        model: [{ type: Input, args: ['model',] }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        label: [{ type: Input }],
        placeholder: [{ type: Input }],
        options: [{ type: Input }],
        url: [{ type: Input }],
        filter: [{ type: Input }],
        limit: [{ type: Input }],
        errors: [{ type: Input }],
        addItemLabel: [{ type: Input }],
        ngModelChange: [{ type: Output }]
    };
    return InputMultiSelectTwoWayBindingComponent;
}());
export { InputMultiSelectTwoWayBindingComponent };
if (false) {
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.error;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.required;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.readonly;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.placeholder;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.options;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.url;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.filter;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.limit;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.addItemLabel;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.ngModelChange;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.guid;
    /** @type {?} */
    InputMultiSelectTwoWayBindingComponent.prototype.value;
}
