// anef-ui/esm5/lib/forms/input-switch/input-switch.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-switch/input-switch.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { FormGroup } from '@angular/forms';
var InputSwitchComponent = /** @class */ (function () {
    function InputSwitchComponent() {
        this.customStyleClass = 'anef-switch';
        this.change = new EventEmitter();
    }
    /**
     * @return {?}
     */
    InputSwitchComponent.prototype.isChecked = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].value;
    };
    /**
     * @return {?}
     */
    InputSwitchComponent.prototype.isDisabled = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].disabled;
    };
    /**
     * @return {?}
     */
    InputSwitchComponent.prototype.formContainsError = /**
     * @return {?}
     */
    function () {
        return this.formGroup.controls[this.controlName].errors !== null &&
            (this.formGroup.controls[this.controlName].dirty || this.formGroup.controls[this.controlName].touched);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    InputSwitchComponent.prototype.onchange = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (this.change) {
            this.change.emit(event);
        }
    };
    InputSwitchComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-switch',
                    template: "<div *ngIf=[formGroup] [formGroup]=\"formGroup\">\n  <label *ngIf=\"label\" class=\"switch-position\"> {{label}} :</label>\n  <span *ngIf=\"choice1\" class=\"switch-position\"\n    [ngClass]='{\"ui-label-selected\": !isChecked(), \"ui-label-disabled\": isDisabled()}'> {{choice1}}</span>\n  <p-inputSwitch [formControlName]=\"controlName\" (onChange)=\"onchange($event)\" [styleClass]=\"customStyleClass\">\n  </p-inputSwitch>\n  <span *ngIf=\"choice2\" class=\"switch-position\"\n    [ngClass]='{\"ui-label-selected\": isChecked(), \"ui-label-disabled\": isDisabled()}'> {{choice2}}</span>\n\n</div>",
                    styles: [":host ::ng-deep .ui-inputswitch .ui-inputswitch-slider:before{background-color:#256cb9}:host ::ng-deep .anef-switch.ui-inputswitch:hover .ui-inputswitch-slider{background-color:#94a9c2}:host ::ng-deep .anef-switch.ui-inputswitch .ui-inputswitch-slider{background-color:#aac5e3}:host ::ng-deep .anef-switch.ui-inputswitch.ui-inputswitch-checked.ui-inputswitch-focus .ui-inputswitch-slider{background:#aac5e3}:host ::ng-deep .anef-switch.ui-inputswitch.ui-inputswitch-checked:hover .ui-inputswitch-slider{background:#94a9c2}:host ::ng-deep .ui-state-disabled,:host ::ng-deep .ui-widget:disabled{opacity:.5!important;background-color:transparent}span.ui-label-selected{font-weight:700}span.ui-label-disabled{color:#c7c7c7}.switch-position{padding-left:.4em;padding-right:.4em;position:relative;top:-6px}"]
                }] }
    ];
    InputSwitchComponent.propDecorators = {
        formGroup: [{ type: Input }],
        controlName: [{ type: Input }],
        choice1: [{ type: Input }],
        choice2: [{ type: Input }],
        customStyleClass: [{ type: Input }],
        label: [{ type: Input }],
        change: [{ type: Output }]
    };
    return InputSwitchComponent;
}());
export { InputSwitchComponent };
if (false) {
    /** @type {?} */
    InputSwitchComponent.prototype.formGroup;
    /** @type {?} */
    InputSwitchComponent.prototype.controlName;
    /** @type {?} */
    InputSwitchComponent.prototype.choice1;
    /** @type {?} */
    InputSwitchComponent.prototype.choice2;
    /** @type {?} */
    InputSwitchComponent.prototype.customStyleClass;
    /** @type {?} */
    InputSwitchComponent.prototype.label;
    /** @type {?} */
    InputSwitchComponent.prototype.change;
}
