// anef-ui/esm5/lib/forms/input-multi-select/input-multi-select.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-multi-select/input-multi-select.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// Angular
import { Component, Input } from '@angular/core';
import { FormGroup } from '@angular/forms';
/**
 * @record
 */
export function MultiSelectOption() { }
if (false) {
    /** @type {?} */
    MultiSelectOption.prototype.label;
    /** @type {?} */
    MultiSelectOption.prototype.value;
}
var InputMultiSelectComponent = /** @class */ (function () {
    function InputMultiSelectComponent() {
        this.filter = false;
        this.maxSelectedItemsLabel = '{0} éléments sélectionnés';
        this.defaultLabel = 'Choisir';
    }
    InputMultiSelectComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'input-multi-select',
                    template: "<div class=\"p-grid nogutter\" *ngIf=[formGroup] [formGroup]=\"formGroup\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" class=\"multi-select-label\" [htmlFor]=\"controlName\">\n      {{label}} :\n    </label>\n\n    <p-multiSelect\n      [filter]=\"filter\"\n      [defaultLabel]=\"defaultLabel\"\n      [selectedItemsLabel]=\"maxSelectedItemsLabel\"\n      [options]=\"option\"\n      [formControlName]=\"controlName\"\n      class=\"multiselect-custom\">\n      <ng-template let-item pTemplate=\"item\">\n        <div class=\"item-value\">\n          <i *ngIf=\"item.value.icon\" [ngClass]=item.value.icon></i>\n          <div> {{item.value.name}}</div>\n        </div>\n      </ng-template>\n    </p-multiSelect>\n  </div>\n</div>\n",
                    styles: ["label{color:#656565;font-size:1.4rem}:host ::ng-deep .p-multiselect{min-width:15rem}:host ::ng-deep .multiselect-custom-virtual-scroll .p-multiselect{min-width:20rem}:host ::ng-deep .multiselect-custom .p-multiselect-label{padding-top:.25rem;padding-bottom:.25rem}:host ::ng-deep .multiselect-custom .item-value{padding:.25rem .5rem;border-radius:3px;display:inline-flex;margin-right:.5rem;background-color:var(--primary-color);color:var(--primary-color-text)}:host ::ng-deep .multiselect-custom .item-value i{padding-right:8px;padding-top:2px}"]
                }] }
    ];
    InputMultiSelectComponent.propDecorators = {
        formGroup: [{ type: Input }],
        controlName: [{ type: Input }],
        label: [{ type: Input }],
        option: [{ type: Input }],
        filter: [{ type: Input }],
        maxSelectedItemsLabel: [{ type: Input }],
        defaultLabel: [{ type: Input }]
    };
    return InputMultiSelectComponent;
}());
export { InputMultiSelectComponent };
if (false) {
    /** @type {?} */
    InputMultiSelectComponent.prototype.formGroup;
    /** @type {?} */
    InputMultiSelectComponent.prototype.controlName;
    /** @type {?} */
    InputMultiSelectComponent.prototype.label;
    /** @type {?} */
    InputMultiSelectComponent.prototype.option;
    /** @type {?} */
    InputMultiSelectComponent.prototype.filter;
    /** @type {?} */
    InputMultiSelectComponent.prototype.maxSelectedItemsLabel;
    /** @type {?} */
    InputMultiSelectComponent.prototype.defaultLabel;
}
