// anef-ui/esm5/lib/success-modal/success-modal.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/success-modal/success-modal.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
var SuccessModalComponent = /** @class */ (function () {
    function SuccessModalComponent() {
        this.displaySuccessChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    SuccessModalComponent.prototype.onClose = /**
     * @return {?}
     */
    function () {
        this.displaySuccessChange.emit(false);
    };
    /**
     * @return {?}
     */
    SuccessModalComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.displaySuccessChange.unsubscribe();
    };
    SuccessModalComponent.decorators = [
        { type: Component, args: [{
                    selector: 'lib-success-modal',
                    template: "<p-dialog\n  [(visible)]=\"displaySuccess\"\n  [modal]=\"true\"\n  [closable]=\"false\"\n  [responsive]=\"true\"\n  (onAfterHide)=\"onClose()\"\n  [style]=\"{width: '60rem', minWidth: '28rem'}\" [minY]=\"70\"\n  [baseZIndex]=\"10000\">\n  <p-header>\n    {{ title }}\n  </p-header>\n  <div class=\"msgBox info\">\n    <span class=\"fa fa-check-square-o check-icon\"></span>\n    <p>{{titleCheck}}</p>\n  </div>\n  <p class=\"bold\">{{textBold | trim}}.</p>\n  <p>{{text | trim}}.</p>\n  <p *ngIf=\"facultativeText\">{{facultativeText}}</p>\n  <p-footer>\n    <button type=\"button\" pButton (click)=\"onClose()\" label=\"{{labelBtn}}\"></button>\n  </p-footer>\n</p-dialog>\n",
                    styles: [".check-icon{color:#0c6b0d!important}"]
                }] }
    ];
    /** @nocollapse */
    SuccessModalComponent.ctorParameters = function () { return []; };
    SuccessModalComponent.propDecorators = {
        displaySuccess: [{ type: Input }],
        title: [{ type: Input }],
        titleCheck: [{ type: Input }],
        text: [{ type: Input }],
        facultativeText: [{ type: Input }],
        labelBtn: [{ type: Input }],
        textBold: [{ type: Input }],
        displaySuccessChange: [{ type: Output }]
    };
    return SuccessModalComponent;
}());
export { SuccessModalComponent };
if (false) {
    /** @type {?} */
    SuccessModalComponent.prototype.displaySuccess;
    /** @type {?} */
    SuccessModalComponent.prototype.title;
    /** @type {?} */
    SuccessModalComponent.prototype.titleCheck;
    /** @type {?} */
    SuccessModalComponent.prototype.text;
    /** @type {?} */
    SuccessModalComponent.prototype.facultativeText;
    /** @type {?} */
    SuccessModalComponent.prototype.labelBtn;
    /** @type {?} */
    SuccessModalComponent.prototype.textBold;
    /** @type {?} */
    SuccessModalComponent.prototype.displaySuccessChange;
}
