// anef-ui/esm5/lib/side-bar/side-bar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/side-bar/side-bar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
/** @enum {string} */
var SideBarType = {
    DOCUMENTS: "DOCUMENTS",
    APPOINTMENT: "APPOINTMENT",
    MESSAGESINTERN: "MESSAGESINTERN",
    MESSAGES: "MESSAGES",
    HISTORIQUE: "HISTORIQUE",
    LIBRARY: "LIBRARY",
};
export { SideBarType };
/**
 * @record
 */
export function SideMenuItem() { }
if (false) {
    /** @type {?|undefined} */
    SideMenuItem.prototype.icon;
    /** @type {?|undefined} */
    SideMenuItem.prototype.title;
    /** @type {?} */
    SideMenuItem.prototype.module;
    /** @type {?|undefined} */
    SideMenuItem.prototype.content;
}
var SideBarComponent = /** @class */ (function () {
    function SideBarComponent() {
        this.activeIndexChange = new EventEmitter();
        this.SideBarType = SideBarType;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    SideBarComponent.prototype.handleChange = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.activeIndex = event.index;
        this.activeIndexChange.emit(this.activeIndex);
    };
    /**
     * @param {?} section
     * @return {?}
     */
    SideBarComponent.prototype.getIconClass = /**
     * @param {?} section
     * @return {?}
     */
    function (section) {
        switch (section) {
            case SideBarType.DOCUMENTS: {
                return 'fa fa-fw fa-paperclip';
            }
            case SideBarType.APPOINTMENT: {
                return 'fa fa-fw fa-calendar';
            }
            case SideBarType.MESSAGESINTERN: {
                return 'fa fa-fw fa-comments-o';
            }
            case SideBarType.MESSAGES: {
                return 'fa fa-fw fa-envelope';
            }
            case SideBarType.HISTORIQUE: {
                return 'fa fa-fw fa-history';
            }
            case SideBarType.LIBRARY: {
                return 'fa fa-fw fa-book';
            }
        }
    };
    SideBarComponent.decorators = [
        { type: Component, args: [{
                    selector: 'anef-side-bar',
                    template: "<p-tabView  orientation=\"left\" class=\"anef-sidebar\" [activeIndex]=\"activeIndex\" (onChange)=\"handleChange($event)\">\n  <p-tabPanel leftIcon=\"{{ getIconClass(section.module) }}\" *ngFor=\"let section of sections; let first = first;\" [selected]=\"first\">\n    <ng-content *ngIf=\"section.module === SideBarType.DOCUMENTS\" select=\"[documents]\"></ng-content>\n    <ng-content *ngIf=\"section.module === SideBarType.APPOINTMENT\" select=\"[appointment]\"></ng-content>\n    <ng-content *ngIf=\"section.module === SideBarType.MESSAGESINTERN\" select=\"[intern-messages]\"></ng-content>\n    <ng-content *ngIf=\"section.module === SideBarType.MESSAGES\" select=\"[messages]\"></ng-content>\n    <ng-content *ngIf=\"section.module === SideBarType.HISTORIQUE\" select=\"[historique]\"></ng-content>\n    <ng-content *ngIf=\"section.module === SideBarType.LIBRARY\" select=\"[library]\"></ng-content>\n  </p-tabPanel>\n</p-tabView>\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    SideBarComponent.ctorParameters = function () { return []; };
    SideBarComponent.propDecorators = {
        sections: [{ type: Input }],
        activeIndex: [{ type: Input }],
        activeIndexChange: [{ type: Output }]
    };
    return SideBarComponent;
}());
export { SideBarComponent };
if (false) {
    /** @type {?} */
    SideBarComponent.prototype.sections;
    /** @type {?} */
    SideBarComponent.prototype.activeIndex;
    /** @type {?} */
    SideBarComponent.prototype.activeIndexChange;
    /** @type {?} */
    SideBarComponent.prototype.SideBarType;
}
