// anef-ui/esm5/lib/forms/input-switch-two-way-binding/input-switch-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-switch-two-way-binding/input-switch-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
var InputSwitchTwoWayBindingComponent = /** @class */ (function () {
    function InputSwitchTwoWayBindingComponent() {
        this.customStyleClass = 'anef-switch';
        this.change = new EventEmitter();
    }
    /**
     * @return {?}
     */
    InputSwitchTwoWayBindingComponent.prototype.isChecked = /**
     * @return {?}
     */
    function () {
        return this.model;
    };
    /**
     * @return {?}
     */
    InputSwitchTwoWayBindingComponent.prototype.isDisabled = /**
     * @return {?}
     */
    function () {
        return this.disabled;
    };
    /**
     * @param {?} event
     * @return {?}
     */
    InputSwitchTwoWayBindingComponent.prototype.onchange = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (this.change) {
            this.change.emit(event);
        }
    };
    InputSwitchTwoWayBindingComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-switch-two-way-binding',
                    template: "<div>\n  <label *ngIf=\"label\" class=\"switch-position\"> {{label}} :</label>\n  <span *ngIf=\"choice1\" class=\"switch-position\"\n    [ngClass]='{\"ui-label-selected\": !isChecked(), \"ui-label-disabled\": isDisabled()}'> {{choice1}}</span>\n  <p-inputSwitch [(ngModel)]=\"model\" [disabled]=\"disabled\" (onChange)=\"onchange($event)\"\n    [styleClass]=\"customStyleClass\"></p-inputSwitch>\n  <span *ngIf=\"choice2\" class=\"switch-position\"\n    [ngClass]='{\"ui-label-selected\": isChecked(), \"ui-label-disabled\": isDisabled()}'> {{choice2}}</span>\n\n</div>\n",
                    styles: [":host ::ng-deep .ui-inputswitch .ui-inputswitch-slider:before{background-color:#256cb9}:host ::ng-deep .anef-switch.ui-inputswitch:hover .ui-inputswitch-slider{background-color:#94a9c2}:host ::ng-deep .anef-switch.ui-inputswitch .ui-inputswitch-slider{background-color:#aac5e3}:host ::ng-deep .anef-switch.ui-inputswitch.ui-inputswitch-checked.ui-inputswitch-focus .ui-inputswitch-slider{background:#aac5e3}:host ::ng-deep .anef-switch.ui-inputswitch.ui-inputswitch-checked:hover .ui-inputswitch-slider{background:#94a9c2}:host ::ng-deep .ui-state-disabled,:host ::ng-deep .ui-widget:disabled{opacity:.5!important;background-color:transparent}span.ui-label-selected{font-weight:700}span.ui-label-disabled{color:#c7c7c7}.switch-position{padding-left:.4em;padding-right:.4em;position:relative;top:-6px}"]
                }] }
    ];
    InputSwitchTwoWayBindingComponent.propDecorators = {
        model: [{ type: Input }],
        disabled: [{ type: Input }],
        choice1: [{ type: Input }],
        choice2: [{ type: Input }],
        customStyleClass: [{ type: Input }],
        label: [{ type: Input }],
        change: [{ type: Output }]
    };
    return InputSwitchTwoWayBindingComponent;
}());
export { InputSwitchTwoWayBindingComponent };
if (false) {
    /** @type {?} */
    InputSwitchTwoWayBindingComponent.prototype.model;
    /** @type {?} */
    InputSwitchTwoWayBindingComponent.prototype.disabled;
    /** @type {?} */
    InputSwitchTwoWayBindingComponent.prototype.choice1;
    /** @type {?} */
    InputSwitchTwoWayBindingComponent.prototype.choice2;
    /** @type {?} */
    InputSwitchTwoWayBindingComponent.prototype.customStyleClass;
    /** @type {?} */
    InputSwitchTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputSwitchTwoWayBindingComponent.prototype.change;
}
