// anef-ui/esm5/lib/forms/input-radio/input-radio.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-radio/input-radio.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// Angular
import { Component, Input } from '@angular/core';
import { FormGroup } from '@angular/forms';
var InputRadioComponent = /** @class */ (function () {
    function InputRadioComponent() {
        this.required = false;
        this.readonly = false;
        this.column = false;
        this.spread = false;
        this.messages = {
            required: 'Ce champ est obligatoire'
        };
        this.defaultDisplay = 'N/A';
        this.classStyle = false;
    }
    /**
     * @return {?}
     */
    InputRadioComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (!this.name) {
            this.name = this.controlName;
        }
    };
    /**
     * @return {?}
     */
    InputRadioComponent.prototype.formContainsError = /**
     * @return {?}
     */
    function () {
        return (!!this.formGroup &&
            !!this.formGroup.controls[this.controlName] &&
            this.formGroup.controls[this.controlName].errors !== null &&
            (this.formGroup.controls[this.controlName].dirty || this.formGroup.controls[this.controlName].touched));
    };
    /**
     * @return {?}
     */
    InputRadioComponent.prototype.toString = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var found = null;
        if (this.formGroup && this.controlName && this.options) {
            /** @type {?} */
            var value_1 = this.formGroup.controls[this.controlName].value;
            found = (this.options && this.options.filter((/**
             * @param {?} option
             * @return {?}
             */
            function (option) { return option.value === value_1; }))[0]) || null;
        }
        return (found && found.label) || this.defaultDisplay;
    };
    InputRadioComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-radio',
                    template: "<div class=\"p-grid nogutter\" *ngIf=\"formGroup\" [formGroup]=\"formGroup\">\n  <div class=\"p-col-12\">\n    <fieldset>\n      <legend *ngIf=\"label\" class=\"radio-label\">\n          {{label}} <span class=\"redish\" *ngIf=\"required\">*</span> :\n      </legend>\n      <ng-content select=\"[sub-label]\"></ng-content>\n\n      <div *ngIf=\"!readonly; else textual\"\n          class=\"p-grid nogutter\" [ngClass]=\"{'p-dir-col': column}\">\n        <ng-container *ngFor=\"let option of options\">\n          <div class=\"p-col\" [ngClass]=\"{'no-col': !spread, 'classStyle': classStyle}\">\n            <p-radioButton [name]=\"name\"\n                          [value]=\"option.value\"\n                          [label]=\"option.label\"\n                          [formControlName]=\"controlName\"\n                          [styleClass]=\"option.styleClass\">\n            </p-radioButton>\n          </div>\n        </ng-container>\n      </div>\n    </fieldset>\n    <ng-template #textual>\n      <span class=\"textual\">{{toString()}}</span>\n    </ng-template>\n\n    <ng-content select=\"[complement]\"></ng-content>\n\n    <div class=\"redish errors\" *ngIf=\"formContainsError()\">\n      <div *ngIf=\"formGroup.controls[controlName].errors.required\">\n        <i class=\"fa fa-exclamation-circle\" aria-hidden=\"true\"></i>\n        <span> {{messages.required}} </span>\n      </div>\n\n      <ng-content select=\"[messages]\"></ng-content>\n    </div>\n  </div>\n</div>\n",
                    styles: ["label{width:100%}.no-col{flex-grow:initial;flex-basis:initial}.textual{display:block}fieldset,legend{margin:0;padding:0;border:0}fieldset legend{border-bottom:0}.classStyle{padding-right:2rem}"]
                }] }
    ];
    InputRadioComponent.propDecorators = {
        formGroup: [{ type: Input }],
        controlName: [{ type: Input }],
        name: [{ type: Input }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        label: [{ type: Input }],
        column: [{ type: Input }],
        spread: [{ type: Input }],
        options: [{ type: Input }],
        messages: [{ type: Input }],
        defaultDisplay: [{ type: Input }],
        classStyle: [{ type: Input }]
    };
    return InputRadioComponent;
}());
export { InputRadioComponent };
if (false) {
    /** @type {?} */
    InputRadioComponent.prototype.formGroup;
    /** @type {?} */
    InputRadioComponent.prototype.controlName;
    /** @type {?} */
    InputRadioComponent.prototype.name;
    /** @type {?} */
    InputRadioComponent.prototype.required;
    /** @type {?} */
    InputRadioComponent.prototype.readonly;
    /** @type {?} */
    InputRadioComponent.prototype.label;
    /** @type {?} */
    InputRadioComponent.prototype.column;
    /** @type {?} */
    InputRadioComponent.prototype.spread;
    /** @type {?} */
    InputRadioComponent.prototype.options;
    /** @type {?} */
    InputRadioComponent.prototype.messages;
    /** @type {?} */
    InputRadioComponent.prototype.defaultDisplay;
    /** @type {?} */
    InputRadioComponent.prototype.classStyle;
}
