// anef-ui/esm2015/lib/forms/input-radio-two-way-binding/input-radio-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-radio-two-way-binding/input-radio-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { generateGuid } from '../../shared/guid';
export class InputRadioTwoWayBindingComponent {
    constructor() {
        this.required = false;
        this.disabled = false;
        this.column = false;
        this.spread = false;
        this.ngModelChange = new EventEmitter();
        this.guid = generateGuid();
        this.error = null;
    }
    /**
     * @param {?} errors
     * @return {?}
     */
    set errors(errors) {
        if (!errors) {
            this.error = null;
        }
        else if (typeof (errors) === 'string') {
            this.error = [errors];
        }
        else {
            this.error = errors;
        }
    }
}
InputRadioTwoWayBindingComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line: component-selector
                selector: 'anef-input-radio-two-way-binding',
                template: "<div class=\"p-grid nogutter\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" class=\"radio-label\" for=\"{{guid}}\">\n      {{label}} <span *ngIf=\"required && !disabled\" class=\"redish\">*</span> :\n    </label>\n\n    <div class=\"input\">\n      <div [ngClass]=\"{'p-dir-col': column}\" class=\"p-grid nogutter\">\n        <ng-container *ngFor=\"let option of options\">\n          <div [ngClass]=\"{'no-col': !spread}\" class=\"p-col\">\n            <p-radioButton (ngModelChange)=\"ngModelChange.emit($event)\"\n                           [(ngModel)]=\"model\"\n                           [disabled]=\"disabled\"\n                           [label]=\"option.label\"\n                           [ngClass]=\"{'error': error && error.length}\"\n                           [styleClass]=\"option.styleClass\"\n                           [value]=\"option.value\"\n                           name=\"{{guid}}\">\n            </p-radioButton>\n          </div>\n        </ng-container>\n      </div>\n    </div>\n\n    <div *ngIf=\"error && error.length\" class=\"error redish\">\n      <div *ngFor=\"let errorLine of error\">\n        <p>\n          <i class=\"pi pi-info-circle\"></i>\n          <span>{{errorLine}}</span>\n        </p>\n      </div>\n    </div>\n\n  </div>\n</div>\n",
                styles: ["label{width:100%}.no-col{flex-grow:initial;flex-basis:initial}.textual{display:block}"]
            }] }
];
InputRadioTwoWayBindingComponent.propDecorators = {
    model: [{ type: Input }],
    name: [{ type: Input }],
    required: [{ type: Input }],
    disabled: [{ type: Input }],
    label: [{ type: Input }],
    column: [{ type: Input }],
    spread: [{ type: Input }],
    options: [{ type: Input }],
    ngModelChange: [{ type: Output }],
    errors: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.model;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.name;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.required;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.disabled;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.column;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.spread;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.options;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.ngModelChange;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.guid;
    /** @type {?} */
    InputRadioTwoWayBindingComponent.prototype.error;
}
