// anef-ui/esm5/lib/forms/input-text-two-way-binding/input-text-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-text-two-way-binding/input-text-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { generateGuid } from '../../shared/guid';
/**
 * @record
 */
function Icon() { }
if (false) {
    /** @type {?} */
    Icon.prototype.type;
    /** @type {?} */
    Icon.prototype.class;
}
var InputTextTwoWayBindingComponent = /** @class */ (function () {
    function InputTextTwoWayBindingComponent() {
        this.MAX_LENGTH = 524288;
        this.error = null;
        this.type = 'text';
        this.required = false;
        this.readonly = false;
        this.capitalize = false;
        this.maxLength = this.MAX_LENGTH;
        this.disabled = false;
        this.ngModelChange = new EventEmitter();
        this.guid = generateGuid();
    }
    Object.defineProperty(InputTextTwoWayBindingComponent.prototype, "errors", {
        set: /**
         * @param {?} errors
         * @return {?}
         */
        function (errors) {
            if (!errors) {
                this.error = null;
            }
            else if (typeof (errors) === 'string') {
                this.error = [errors];
            }
            else {
                this.error = errors;
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} event
     * @return {?}
     */
    InputTextTwoWayBindingComponent.prototype.onModelChange = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (this.capitalize) {
            event = event.charAt(0).toUpperCase() + event.slice(1);
        }
        this.ngModelChange.emit(event);
    };
    /**
     * @return {?}
     */
    InputTextTwoWayBindingComponent.prototype.toString = /**
     * @return {?}
     */
    function () {
        return this.model ? this.model : '-';
    };
    InputTextTwoWayBindingComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-text-two-way-binding',
                    template: "<div class=\"p-grid nogutter\">\n  <div class=\"p-col-12\">\n    <label *ngIf=\"label\" for=\"{{guid}}\">\n      {{label}} <span *ngIf=\"required && !readonly\" class=\"redish\">*</span> :\n    </label>\n\n    <div class=\"input\">\n      <div *ngIf=\"!readonly; else textual\" class=\"ui-inputgroup\">\n        <input (ngModelChange)=\"onModelChange($event)\"\n               [(ngModel)]=\"model\"\n               [disabled]=\"disabled\"\n               [maxlength]=maxLength\n               [ngClass]=\"{'error': error && error.length}\"\n               [readonly]=\"readonly\"\n               [required]=\"required\"\n               id=\"{{guid}}\"\n               name=\"{{guid}}\"\n               pInputText\n               placeholder=\"{{placeholder}}\"\n               type=\"{{type}}\">\n        <button *ngIf=\"icon\" [class]=\"icon.class\" [icon]=\"icon.type\" pButton type=\"button\"></button>\n      </div>\n\n      <ng-template #textual>\n        <div class=\"textual\">{{ toString() }}</div>\n      </ng-template>\n    </div>\n\n    <div *ngIf=\"error && error.length\" class=\"error redish\">\n      <div *ngFor=\"let errorLine of error\">\n        <p>\n          <i class=\"pi pi-info-circle\"></i>\n          <span>{{errorLine}}</span>\n        </p>\n      </div>\n    </div>\n\n  </div>\n</div>\n",
                    styles: ["label{color:#656565;font-size:1.4rem}.textual{margin-top:7px}"]
                }] }
    ];
    InputTextTwoWayBindingComponent.propDecorators = {
        model: [{ type: Input }],
        type: [{ type: Input }],
        required: [{ type: Input }],
        readonly: [{ type: Input }],
        capitalize: [{ type: Input }],
        label: [{ type: Input }],
        placeholder: [{ type: Input }],
        maxLength: [{ type: Input }],
        errors: [{ type: Input }],
        icon: [{ type: Input }],
        disabled: [{ type: Input }],
        ngModelChange: [{ type: Output }]
    };
    return InputTextTwoWayBindingComponent;
}());
export { InputTextTwoWayBindingComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    InputTextTwoWayBindingComponent.prototype.MAX_LENGTH;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.error;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.model;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.type;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.required;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.readonly;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.capitalize;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.placeholder;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.maxLength;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.icon;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.disabled;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.ngModelChange;
    /** @type {?} */
    InputTextTwoWayBindingComponent.prototype.guid;
}
