// anef-ui/esm5/lib/forms/input-select-button-two-way-binding/input-select-button-two-way-binding.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/forms/input-select-button-two-way-binding/input-select-button-two-way-binding.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { generateGuid } from '../../shared/guid';
var InputSelectButtonTwoWayBindingComponent = /** @class */ (function () {
    function InputSelectButtonTwoWayBindingComponent() {
        this.error = null;
        this.multiple = false;
        this.ngModelChange = new EventEmitter();
        this.guid = generateGuid();
    }
    Object.defineProperty(InputSelectButtonTwoWayBindingComponent.prototype, "errors", {
        set: /**
         * @param {?} errors
         * @return {?}
         */
        function (errors) {
            if (!errors) {
                this.error = null;
            }
            else if (typeof (errors) === 'string') {
                this.error = [errors];
            }
            else {
                this.error = errors;
            }
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    InputSelectButtonTwoWayBindingComponent.prototype.toString = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (!this.options) {
            return '-';
        }
        /** @type {?} */
        var selectedOption = this.options.find((/**
         * @param {?} option
         * @return {?}
         */
        function (option) { return _this.model === option.value; }));
        return selectedOption ? selectedOption.label : '-';
    };
    InputSelectButtonTwoWayBindingComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line: component-selector
                    selector: 'anef-input-select-button-two-way-binding',
                    template: "<label *ngIf=\"label\" for=\"{{guid}}\">\n  {{label}} <span *ngIf=\"required && !readonly\" class=\"redish\">*</span> :\n</label>\n\n<div class=\"input\">\n  <div *ngIf=\"!readonly; else textual\" class=\"p-grid\">\n    <p-selectButton\n      (ngModelChange)=\"ngModelChange.emit($event)\"\n      [(ngModel)]=\"model\"\n      [multiple]=\"multiple\"\n      [ngClass]=\"{'error': error && error.length}\"\n      [options]=\"options\"\n      class=\"custom-input-select-button p-col-12\"\n      id=\"{{guid}}\"\n      name=\"{{guid}}\">\n    </p-selectButton>\n  </div>\n\n  <ng-template #textual>\n    <div>{{ toString() }}</div>\n  </ng-template>\n</div>\n\n<div *ngIf=\"error && error.length\" class=\"error redish\">\n  <div *ngFor=\"let errorLine of error\">\n    <p>\n      <i class=\"pi pi-info-circle\"></i>\n      <span>{{errorLine}}</span>\n    </p>\n  </div>\n</div>\n",
                    styles: ["label{color:#656565;font-size:1.4rem}.custom-input-select-button{margin-top:-.8rem}"]
                }] }
    ];
    InputSelectButtonTwoWayBindingComponent.propDecorators = {
        model: [{ type: Input }],
        readonly: [{ type: Input }],
        required: [{ type: Input }],
        label: [{ type: Input }],
        multiple: [{ type: Input }],
        options: [{ type: Input }],
        errors: [{ type: Input }],
        ngModelChange: [{ type: Output }]
    };
    return InputSelectButtonTwoWayBindingComponent;
}());
export { InputSelectButtonTwoWayBindingComponent };
if (false) {
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.error;
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.model;
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.readonly;
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.required;
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.label;
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.multiple;
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.options;
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.ngModelChange;
    /** @type {?} */
    InputSelectButtonTwoWayBindingComponent.prototype.guid;
}
